contracts/BasicRandomizerV2.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Creatd By: Art Blocks Inc.
+
+pragma solidity 0.8.9;
+
+import "./interfaces/0.8.x/IRandomizerV2.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+
+contract BasicRandomizerV2 is IRandomizerV2, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    function assignCoreAndRenounce(address _genArt721Core) external onlyOwner {
+        renounceOwnership();
+        genArt721Core = IGenArt721CoreContractV3(_genArt721Core);
+    }
+
+    // When `genArt721Core` calls this, it can be assured that the randomizer
+    // will set a bytes32 hash for tokenId `_tokenId` on the core contract.
+    function assignTokenHash(uint256 _tokenId) external {
+        uint256 time = block.timestamp;
```

Comment:
I think this should have an assertion that the `msg.sender` is `genArt721Core` ya?

Reply:
Ah nevermind, I see we have the assertion in `setTokenHash_8PT`. That said, probably worth a dev documentation comment making this assumption by the randomizer and explicit one.

Reply:
no you are correct - this does need that assertion. I implemented the fix last night but forgot to push the commit! updated!

Reply:
Ah yeah the assertion definitely does need to go both ways 😅 

I probably shouldn't review PRs before coffee. TY for updating!